ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/operations/BeginSubOperationTest.java

```diff
@@ -95,6 +94,6 @@ public void shouldHaltWithInvalidSubRoutineEntryWhenBeginSubIsExecuted() {
     frame.setPC(CURRENT_PC);
     final OperationResult result = operation.execute(frame, null);
 
-    assertThat(result.getHaltReason()).contains(ExceptionalHaltReason.INVALID_SUB_ROUTINE_ENTRY);
+    assertThat(result.getHaltReason()).isNotEmpty();
```

Comment:
I don't understand why this test had to be modified 🤔

Reply:
The test would also fail because of out of gas, and we check gas sooner now.  I think your intent is right in that the only way this should fail is invalid subroutine entry, so I re-worked the test to provide more gas and only accept that exceptional halt.